libs/data/bbhnet/data/utils.py

```diff
@@ -66,14 +69,31 @@ def sample_kernels(
     # always make sure that the center of x's
     # 1st axis is in the kernel that we sample
     # if we're doing >1D sampling
-    min_sample_start = max(
-        x.shape[-1] // 2 + 1 - trigger_distance_size - (size / 2), 0
-    )
-    max_sample_start = min(
-        x.shape[-1] // 2 - 1 - (size / 2) + trigger_distance_size,
-        x.shape[-1] - size,
-    )
 
+    # TODO: not sure if theres a logical way to
+    # combine this if else statement.
+    # Was spending too much time thinking about it
+
+    # allows for kernels that don't contain trigger
+    if trigger_distance_size > 0:
+        min_sample_start = max(
+            x.shape[-1] // 2 + 1 - trigger_distance_size - size, 0
+        )
+        max_sample_start = min(
+            x.shape[-1] // 2 - 1,

```

Comment:
This should have a `+ trigger_distance_size` I think, so that you can sample with a leftmost edge that's to the right of `t0`

Reply:
Ah yes I was thinking we enforce that the kernel never goes beyond t0. Of course there is some signal (the ringdown) beyond, butI was envisioning the use case of BNS's where the majority of the signal is _before_ `t0`. If we used `trigger_distance_size` symmetrically, there would be a lot of kernels to the right of `t0` that dont contain any signal. Of course this can be alleviated by incorporating asymmetric `trigger_distance_size` 
 (e.g. `trigger_distance_size_right` and `trigger_distance_size_left`). Let me know what you think.

Reply:
Ahhh of course, it's not symmetric in our case. I think your idea actually makes a lot of sense. The one catch would be that we should make sure to define the "trigger" for glitches the same way: at the _end_ of the glitch, that way this sampling method has the intended behavior for both these types of signals.

Then again, if you think marking cases that contain _just_ ringdown as `1` is valuable, then your proposed asymmetric method might make sense as well.